edgar_tool/text_search.py

```diff
@@ -224,10 +221,14 @@ def _generate_request_args(
         # Add optional parameters
         if entity_id:
             request_args["entityName"] = entity_id
-        if filing_type:
-            request_args["category"] = TEXT_SEARCH_FILING_CATEGORIES_MAPPING[
-                filing_type
-            ]
+        if filing_form and not single_forms:
+            request_args["forms"] = ",".join(
+                TEXT_SEARCH_CATEGORY_FORM_GROUPINGS[filing_form]
+            )
+            request_args["forms"] = request_args["forms"].replace(" ", "-")
+        if filing_form and single_forms:
+            request_args["forms"] = ",".join(single_forms)
+            request_args["forms"] = request_args["forms"].replace(" ", "-")
```

Comment:
This is another place where the behaviour could change to allow giving `single_forms` without `filing_form`. I really like the idea of combining them in the instance that both are provided!

Reply:
I don't get this. You're saying you prefer to have the possibility to run `text_search` with the `single_forms` alone, is that correct?

Reply:
Yes, exactly. I've put a bit more detail on the other comment: https://github.com/bellingcat/EDGAR/pull/25#discussion_r1606816657